### PR TITLE
introduce ruff for lint & format and pixi

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

File: webservices_dispatch_action/env_management.py [hunk 1]
```diff
@@ -13,21 +13,21 @@ class SensitiveEnv:
...
     def __init__(self):
-        self.clasified_info = {}
+        self.classified_info = {}
 
     def hide_env_vars(self):
         """Remove sensitive env vars"""
-        self.clasified_info.update(
+        self.classified_info.update(
             {
-                k: os.environ.pop(k, self.clasified_info.get(k, None))
```
Comment (wolfv) [inline webservices_dispatch_action/env_management.py:22]: One of the automated tools changed this typo - but not sure if that is potentially used externally anywhere.

**Comment by wolfv** [comment]
I am trying to introduce pixi for better lockfile management and safer updates. Also included the usual ruff / mypy combo.

**Review by beckermr — CHANGES_REQUESTED** [review]
Right now, this image gets updated automatically by this GHA job: https://github.com/conda-forge/conda-forge-webservices/blob/main/.github/workflows/clean-and-update.yml when new package versions are released for certain tools.

Are you willing to take on the responsibility of updating the pixi lock file here?

**Comment by wolfv** [comment]
We could automate that with the pixi github action and have a lockfile to be able to roll back to some state?

Isn't it currently just always updating everything on a weekly basis?

I was also hoping to introduce `mypy` in order to catch a certain class of bugs early (renamed functions, etc.).

Currently I am splitting this up and cleaning the Dockerfile.

**Comment by beckermr** [comment]
The webservices code has custom logic for the updates: https://github.com/conda-forge/conda-forge-webservices/blob/main/conda_forge_webservices/update_me.py

**Comment by wolfv** [comment]
My head is spinning a bit, but isn't this action run every week and also pushing a docker image? https://github.com/conda-forge/webservices-dispatch-action/blob/e19a392fd6e7e2d255c3ee76d419fd7d1a6eed93/.github/workflows/tests.yml#L90

**Comment by beckermr** [comment]
Yes. We do both. :)

**Comment by beckermr** [comment]
All of this automation is how we keep ourselves sane when running conda-forge. Otherwise, we'd have to run around updating tools and repos all the time.

**Comment by wolfv** [comment]
The entrypoint is also updating everything when starting?

`conda update --all --yes`

**Comment by beckermr** [comment]
Yep. You need the latest pinnings package for rerendering. At some point I had thought this was the only thing being updated, but then maybe it got switched? I don't recall.

**Pull request closed without merging**